pyfstat/logging.py

```diff
@@ -0,0 +1,136 @@
+"""
+PyFstat's logging implementation.
+
+PyFstat main logger is called `pyfstat` and can be accessed via::
+
+    import logging
+    logger = logging.getLoger('pyfstat')
+
+Basics of logging: For all our purposes, there are *logger* objects
+and *handler* objects. Loggers are the ones in charge of logging,
+hence you call them to emit a logging message with a specific logging
+level (e.g. ``logger.info``); handlers are in charge of redirecting that
+message to a specific place (e.g. a file or your terminal, which is
+usually referred to as a *stream*).
+
+The default behaviour is to attach a ``logging.StreamHandler`` to
+the *pyfstat* logger, printing out to ``sys.stdout`` upon importing the package.
+If, for any reason, ``logging`` cannot access ``sys.stdout`` at import time,
+the exception is reported via ``print`` and no handlers are attached
```

Comment:
Would that actually work - `print` without `stdout`...? :thinking: :slot_machine: 
```
>>> import sys
>>> print("test")
test
>>> sys.stdout=None
>>> print("test")
>>>
```
Well, at least it throws no error, but it also doesn't actually print. ;)

Reply:
I could remove this and let the logger fail...?

Reply:
Not sure if that would be better, as the user then wouldn't have a clear way around that error, because they can't access our setup function without importing first. If they really wanted to get rid of their stdout on purpose, and don't have any logging handlers on their root logger either, let them suffer in silence...?

Also this documentation needs to be updated to clarify that now this default is only done if there are no handlers on the root logger yet, or else those are maintained without extras.